crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -42,3 +53,255 @@ impl Deref for BindGroup {
+/// are generally bound to group 1.
+///
+/// The following field-level attributes are supported:
+/// * `uniform(BINDING_INDEX)`
+///     * The field will be converted to a shader-compatible type using the [`ShaderType`] trait, written to a [`Buffer`], and bound as a uniform.
+///     [`ShaderType`] is implemented for most math types already, such as [`f32`], [`Vec4`](bevy_math::Vec4), and
+///   [`Color`](crate::color::Color). It can also be derived for custom structs.
+/// * `texture(BINDING_INDEX)`
+///     * This field's [`Handle<Image>`](bevy_asset::Handle) will be used to look up the matching [`Texture`](crate::render_resource::Texture)
+///     gpu resource, which will be bound as a texture in shaders. The field will be assumed to implement [`Into<Option<Handle<Image>>>`]. In practice,
+///     most fields should be a [`Handle<Image>`](bevy_asset::Handle) or [`Option<Handle<Image>>`]. If the value of an [`Option<Handle<Image>>`] is
+///     [`None`], the [`FallbackImage`] resource will be used instead. This attribute can be used in conjunction with a `sampler` binding attribute
+///    (with a different binding index) if a binding of the sampler for the [`Image`] is also required.
+/// * `sampler(BINDING_INDEX)`
+///     * This field's [`Handle<Image>`](bevy_asset::Handle) will be used to look up the matching [`Sampler`](crate::render_resource::Sampler) gpu
+///     resource, which will be bound as a sampler in shaders. The field will be assumed to implement [`Into<Option<Handle<Image>>>`]. In practice,
+///     most fields should be a [`Handle<Image>`](bevy_asset::Handle) or [`Option<Handle<Image>>`]. If the value of an [`Option<Handle<Image>>`] is
+///     [`None`], the [`FallbackImage`] resource will be used instead. This attribute can be used in conjunction with a `texture` binding attribute
+///     (with a different binding index) if a binding of the texture for the [`Image`] is also required.
+///
+/// Note that fields without field-level binding attributes will be ignored.
+/// ```
+/// # use bevy_render::{color::Color, render_resource::AsBindGroup};
+/// # use bevy_asset::Handle;
+/// #[derive(AsBindGroup)]
+/// struct CoolMaterial {
+///     #[uniform(0)]
+///     color: Color,
+///     this_field_is_ignored: String,
+/// }
+/// ```
+///
+///  As mentioned above, [`Option<Handle<Image>>`] is also supported:
+/// ```
+/// # use bevy_render::{color::Color, render_resource::AsBindGroup, texture::Image};
+/// # use bevy_asset::Handle;
+/// #[derive(AsBindGroup)]
+/// struct CoolMaterial {
+///     #[uniform(0)]
+///     color: Color,
+///     #[texture(1)]
+///     #[sampler(2)]
+///     color_texture: Option<Handle<Image>>,
+/// }
+/// ```
+/// This is useful if you want a texture to be optional. When the value is [`None`], the [`FallbackImage`] will be used for the binding instead, which defaults
+/// to "pure white".
```

Comment:
When we add support for texture types other than 2D, the fallback image will need to match that type.

Reply:
Makes sense!